ext/zip/php_zip.c

```diff
@@ -2304,8 +2322,8 @@ static ZIPARCHIVE_METHOD(getCommentIndex)
 	}
 
 	PHP_ZIP_STAT_INDEX(intern, index, 0, sb);
-	comment = zip_get_file_comment(intern, index, &comment_len, (int)flags);
-	RETURN_STRINGL((char *)comment, (zend_long)comment_len);
+	comment = zip_file_get_comment(intern, index, &comment_len, (zip_flags_t)flags);
+	RETURN_STRINGL((char *)comment, MIN(comment_len, ZEND_LONG_MAX));
```

Comment:
This MIN doesn't make sense as string lengths are size_t. comment_len should be used directly.

Reply:
Thanks! Done.